JoinColony/colonyCDapp
# Fix: Correctly identify and handle the "Unknown" notifications

## Description

Whenever we create an expenditure (Advanced payment), we basically create 2 kinds of notifications in our Block Ingestor:

1. `NotificationType.Mention` notification via `sendMentionNotifications()` in `expenditureRecipientSet.ts`
2. `NotificationType.ExpenditureReadyForReview` notification via `sendExpenditureUpdateNotifications()` in `expenditureAdded.ts`

The reason why it was saying unknown is because we are simply not handling Mention notifications for Expenditures in the `Notification.tsx` file so all I'm doing is handling it here:

| Before | After |
| ---- | ---- |
| <img width="687" alt="Screenshot 2024-11-05 at 08 45 24" src="https://github.com/user-attachments/assets/fde25f6d-e5b3-4bb7-a5d4-62b699b68d36"> | <img width="688" alt="Screenshot 2024-11-05 at 08 44 47" src="https://github.com/user-attachments/assets/e502d49b-b40c-4ebb-8a64-47efd5a16da7"> |

## Testing

1. Give Amy Owner permissions in General
2. As Leela, create an Advanced Payment using this setup:

<img width="815" alt="image" src="https://github.com/user-attachments/assets/12f42e2e-f7fe-47b9-b8b7-533c4052366a">

3. Log in as Amy
4. Open the Notifications tab
5. Verify that you no longer see the "Unknown" notification item
6. You can also checkout to `master` at this point to compare the result

Resolves #3571
